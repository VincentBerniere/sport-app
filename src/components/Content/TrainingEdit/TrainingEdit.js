import React, { Component } from 'react';
import { Redirect } from 'react-router-dom'
import jsonTraining from '../../../api/training';

class TrainingEdit extends Component {
    state = {
        redirect: false,
        training: {}
    };

    componentDidMount() {
        this.fetchTrainingById();
    }

    fetchTrainingById() {
        jsonTraining.get(`${this.props.match.params.trainingId}/detail`)
            .then(res => {
                if (res.data) {
                    this.setState({ training: res.data });
                }
            });
    }

    updateTraining = event => {
        event.preventDefault();

        jsonTraining.put('', this.state.training)
            .then(res => {
                this.setState({ redirect: true });
            });
    }

    renderRedirect = () => {
        if (this.state.redirect) {
            return <Redirect to={`/trainings/${this.props.match.params.trainingId}/detail`} />
        }
    }

    render() {
        return (
            <form className="ui form" onSubmit={this.updateTraining}>
                <div className="field">
                    <label>Titre</label>
                    <input id="title" type="text" name="title" placeholder="Titre" required maxLength="100" value={this.state.training.title}
                        onChange={event => this.setState({ training: { ...this.state.training, title: event.target.value } })} />
                </div>

                <div className="field">
                    <label>Description</label>
                    <textarea id="description" placeholder="Description" required maxLength="5000" value={this.state.training.description}
                        onChange={event => this.setState({ training: { ...this.state.training, description: event.target.value } })}>
                    </textarea>
                </div>

                <button type="submit" className="ui primary button">Sauvegarder</button>

                {this.renderRedirect()}
            </form>
        );
    }
}

export default TrainingEdit;