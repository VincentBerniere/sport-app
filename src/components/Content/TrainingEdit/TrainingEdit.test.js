import React from 'react';
import { shallow, mount } from 'enzyme'; 
import TrainingEdit from './TrainingEdit';

describe('TrainingEdit', () => {
    let trainingEdit = shallow(<TrainingEdit />);

    it('renders correctly', () => {
        expect(trainingEdit).toMatchSnapshot();
    });

    describe('when typing into the editing form inputs', () => {
        const training = {
            title: 'Training 1',
            description: 'Sed ut perspiciatis unde omnis is ...'
        };

        beforeEach(() => {
            trainingEdit = mount(<TrainingEdit />);
            
            trainingEdit.find('#title').simulate('change', { target: { value: training.title } });
            trainingEdit.find('#description').simulate('change', { target: { value: training.description } });
        });

        it('update the title `state` correctly', () => {
            expect(trainingEdit.state().training.title).toMatchSnapshot(training.title);
        });

        it('update the description `state` correctly', () => {
            expect(trainingEdit.state().training.description).toMatchSnapshot(training.description);
        });
    });
});