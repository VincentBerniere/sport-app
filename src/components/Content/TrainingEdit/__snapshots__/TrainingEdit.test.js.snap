// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainingEdit renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TrainingEdit />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="field"
        >
          <label>
            Titre
          </label>
          <input
            id="title"
            maxLength="100"
            name="title"
            onChange={[Function]}
            placeholder="Titre"
            required={true}
            type="text"
          />
        </div>,
        <div
          className="field"
        >
          <label>
            Description
          </label>
          <textarea
            id="description"
            maxLength="5000"
            onChange={[Function]}
            placeholder="Description"
            required={true}
          />
        </div>,
        <button
          className="ui primary button"
          type="submit"
        >
          Sauvegarder
        </button>,
        undefined,
      ],
      "className": "ui form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Titre
            </label>,
            <input
              id="title"
              maxLength="100"
              name="title"
              onChange={[Function]}
              placeholder="Titre"
              required={true}
              type="text"
            />,
          ],
          "className": "field",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Titre",
            },
            "ref": null,
            "rendered": "Titre",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "title",
              "maxLength": "100",
              "name": "title",
              "onChange": [Function],
              "placeholder": "Titre",
              "required": true,
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Description
            </label>,
            <textarea
              id="description"
              maxLength="5000"
              onChange={[Function]}
              placeholder="Description"
              required={true}
            />,
          ],
          "className": "field",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Description",
            },
            "ref": null,
            "rendered": "Description",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "description",
              "maxLength": "5000",
              "onChange": [Function],
              "placeholder": "Description",
              "required": true,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sauvegarder",
          "className": "ui primary button",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Sauvegarder",
        "type": "button",
      },
      undefined,
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="field"
          >
            <label>
              Titre
            </label>
            <input
              id="title"
              maxLength="100"
              name="title"
              onChange={[Function]}
              placeholder="Titre"
              required={true}
              type="text"
            />
          </div>,
          <div
            className="field"
          >
            <label>
              Description
            </label>
            <textarea
              id="description"
              maxLength="5000"
              onChange={[Function]}
              placeholder="Description"
              required={true}
            />
          </div>,
          <button
            className="ui primary button"
            type="submit"
          >
            Sauvegarder
          </button>,
          undefined,
        ],
        "className": "ui form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Titre
              </label>,
              <input
                id="title"
                maxLength="100"
                name="title"
                onChange={[Function]}
                placeholder="Titre"
                required={true}
                type="text"
              />,
            ],
            "className": "field",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Titre",
              },
              "ref": null,
              "rendered": "Titre",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "title",
                "maxLength": "100",
                "name": "title",
                "onChange": [Function],
                "placeholder": "Titre",
                "required": true,
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Description
              </label>,
              <textarea
                id="description"
                maxLength="5000"
                onChange={[Function]}
                placeholder="Description"
                required={true}
              />,
            ],
            "className": "field",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Description",
              },
              "ref": null,
              "rendered": "Description",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "description",
                "maxLength": "5000",
                "onChange": [Function],
                "placeholder": "Description",
                "required": true,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sauvegarder",
            "className": "ui primary button",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Sauvegarder",
          "type": "button",
        },
        undefined,
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`TrainingEdit when typing into the editing form inputs update the description \`state\` correctly: Sed ut perspiciatis unde omnis is ... 1`] = `undefined`;

exports[`TrainingEdit when typing into the editing form inputs update the title \`state\` correctly: Training 1 1`] = `undefined`;
